### ISSUE TITLE
Not subscribing to Gossip subnets occasionally 

### ISSUE THREAD

**Issue by Nashatyrev** [Nashatyrev/artemis#166]
Sometime a client for some reasons doesn't subscribe to subnets (including columns, attestations, etc).
![image](https://github.com/user-attachments/assets/a1a12e34-c3a0-4e16-bc5e-4363aab6eb7f)

**Comment by Nashatyrev** [comment]
This is how when it works: 

![image](https://github.com/user-attachments/assets/aa25dd6a-33e1-4a10-8242-81b654978abb)

**Comment by Nashatyrev** [comment]
Can be reproduced by delaying network start, and restarting a node after it has been synced: 
```patch
Index: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/ActiveEth2P2PNetwork.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/ActiveEth2P2PNetwork.java b/networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/ActiveEth2P2PNetwork.java
--- a/networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/ActiveEth2P2PNetwork.java	(revision 716c44183b18443289b8b501eac1fe4bdcaacc84)
+++ b/networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/ActiveEth2P2PNetwork.java	(date 1728978663230)
@@ -130,7 +130,9 @@
     // Set the current fork info prior to discovery starting up.
     final ForkInfo currentForkInfo = recentChainData.getCurrentForkInfo().orElseThrow();
     updateForkInfo(currentForkInfo);
-    return super.start().thenAccept(r -> startup());
+    System.err.println("Delaying network startup...");
+    SafeFuture<Void> delayedFuture = asyncRunner.getDelayedFuture(Duration.ofSeconds(5));
+    return super.start().thenCompose(r -> delayedFuture.thenApply(__ -> r)).thenAccept(r -> startup());
   }
 
   private synchronized void startup() {
```

**Comment by Nashatyrev** [comment]
When network start is delayed the sync state changes from `optimistic = true` to `optimistic = false`, but the chain head is not yet updated in the `RecentChainData`. 
When the `ActiveP2PNetwork.startGossip()` is called `RecentChainData.isCurrentHeadOptimistic()` returns `true` and thus no subnet subscriptions are started. It is expected to start when `onSyncStateChanged(optimistic = false)` is called, but it was already called before. 


![image](https://github.com/user-attachments/assets/06899bfb-5c99-4a10-b751-cf8cda357a0c)

**Comment by Nashatyrev** [comment]
Fixed with #167
